### DISCUSSION TITLE
Using the SV Inspector with large VCF files

### DISCUSSION THREAD

**Discussion by john-portwood** [GMOD/jbrowse-components#2103] [Q&A]
Hello! I work with maize, and I have a 4.8GB bgzipped VCF file of maize SNPs. One of my stakeholders would like to view small regions of this data in the SV Inspector, but the file is much too large to load when I try to do so through the inspector's input form. Would it be feasible to implement a method for portions of large files to be viewed in the SV inspector? 

Thank you! I love JBrowse2 and appreciate all of the work you're doing!

**Comment by cmdcolin** [comment]
Hi there
Handling a 4.8Gb file, while this would be awesome, is outside the current capability. 

We could make progress to try to handle somewhat larger files, but the SV inspector is designed to fully load the file into memory...it has to to do this to make the whole-genome overview plot in the circos view, and to do filtering and sorting in the table, and it doesn't do these things in ultra efficient ways currently so even things larger than a hundred Mb could maybe be challenging.

If possible, I would suggest manually subsetting the data using tabix

```
tabix yourfile.vcf.gz region:1-100 > smaller.vcf
```

And then open this smaller file up in the SV inspector. Alternatively, if the file contains for example SNPs and SVs, maybe filtering out to just contain SVs

If you have any example files that we can help debug specifically let us know! Really nice to hear the nice words too, thanks

**Reply by cmdcolin** [reply]
Note: if it is helpful, maybe jbrowse could help perform those subselections in the import form? I just re-read your messsage and saw that you noted "Would it be feasible to implement a method for portions of large files to be viewed in the SV inspector?"
